src/sql_generator.cpp

```diff
@@ -0,0 +1,342 @@
+#include <iostream>
+#include <md_logging.hpp>
+#include <string>
+
+#include "../includes/sql_generator.hpp"
+
+using duckdb::KeywordHelper;
+
+// Utility
+
+std::string table_def::to_string() const {
+  std::ostringstream out;
+  out << KeywordHelper::WriteQuoted(db_name, '"') << "."
+      << KeywordHelper::WriteQuoted(schema_name, '"') << "."
+      << KeywordHelper::WriteQuoted(table_name, '"');
+  return out.str();
+}
+
+void write_joined(
+    std::ostringstream &sql, const std::vector<const column_def *> columns,
+    std::function<void(const std::string &, std::ostringstream &)> print_str) {
+  bool first = true;
+  for (const auto &col : columns) {
+    if (first) {
+      first = false;
+    } else {
+      sql << ", ";
+    }
+    print_str(col->name, sql);
+  }
+}
+
+// DuckDB querying
+// TODO: add test for schema or remove the logic if it's unused
+bool schema_exists(duckdb::Connection &con, const std::string &db_name,
+                   const std::string &schema_name) {
+  std::string query = "SELECT schema_name FROM information_schema.schemata "
```

Comment:
nit: 
```suggestion
  const std::string query = "SELECT schema_name FROM information_schema.schemata "
```
(could even be static)
(same below)